src/Components/Article_votes.jsx

```diff
@@ -0,0 +1,58 @@
+import { useState } from "react";
+import { useParams } from "react-router-dom";
+import { patchVotes } from "../Utils/Api-requests";
+
+const UpvoteDownvoteButtons = ({ votes }) => {
+  const [vote, setVote] = useState(votes);
+  const [upvoteCount, setUpvoteCount] = useState(0);
+  const [downvoteCount, setDownvoteCount] = useState(0);
+  const { article_id } = useParams();
+
+  const handleUpvote = () => {
+    if (upvoteCount === 0) {
+      setUpvoteCount(1);
+      patchVotes(article_id, vote + 1);
+      setVote(vote + 1);
+    } else {
+      setUpvoteCount(0);
+      patchVotes(article_id, vote - 1);
+      setVote(vote - 1);
+    }
+  };
+
+  const handleDownvote = () => {
+    if (downvoteCount === 0) {
+      setDownvoteCount(-1);
+      patchVotes(article_id, vote - 1);
+      setVote(vote - 1);

```

Comment:
If we are using optimistic rendering, we want to show the vote change before the request goes through so that the user gets the instant visual feedback that they have voted. To do this you'd want to swap the order of setVote() and patchVotes,  and then do any error handling for patch votes in a catch block